test/server.ts

```diff
@@ -45,7 +31,65 @@ describe('Onboard server', function () {
           }
         `,
       });
+
       expect(response.data.data.hello).to.be.eq('Hello World!');
     });
   });
+
+  describe('Create user mutation', async function () {
+    const userInput: UserInput = {
+      name: 'Test User',
+      email: 'test@user.com',
+      password: 'password123',
+      birthDate: '2000-01-01',
+    };
+
+    const createUserMutation = {
+      operationName: 'CreateUserMutation',
+      query: `#graphql
+        mutation CreateUserMutation($user: UserInput!) {
+          createUser(user: $user) {
+            id
+            name
+            email
+            birthDate
+          }
+        }
+      `,
+      variables: {
+        user: userInput,
+      },
+    };
+
+    let createdUser: User;
+    let databaseUserInfo: {
+      id: number;
+      name: string;
+      email: string;
+      password: string;
+      birthDate: Date;
+    };
+
+    before(async function () {
+      const response = await axios.post(url, createUserMutation);
+      createdUser = response.data.data.createUser;
+      databaseUserInfo = await prisma.user.findUnique({ where: { email: userInput.email } });
+    });
+
+    it('user ID in response should match user ID in the database', function () {
+      expect(createdUser.id).to.be.eq(databaseUserInfo.id.toString());
+    });
+    it('user name in response should match user name in request', function () {
+      expect(createdUser.name).to.be.eq(userInput.name);
+    });
+    it('user e-mail in response should match user e-mail in request', function () {
+      expect(createdUser.email).to.be.eq(userInput.email);
+    });
+    it('user birth date in response should match user birth date in the request', function () {
+      expect(createdUser.birthDate).to.be.eq(new Date(userInput.birthDate).getTime().toString());
+    });
+    it('user password in request, after encryption, should match user password in the database', function () {
+      expect(bcrypt.compareSync(userInput.password, databaseUserInfo.password)).to.be.equal(true);
+    });
+  });
```

Comment:
Não recomendaria seguir por esse caminho, o alan passou um doc sobre as praticas de testes la no canal, acho que vale dar uma lida, recomendaria refatorar pra cada `it` testar um cenário de forma independente, usando até o `afterEach` para limpar o banco entre um teste e outro.

A motivação é que cada teste seja independente e completo por si só, fora que aqui você só quebrou pra cada it checar um campo, o que não é necessário se considerar que o expect(..) já fala certinho qual parte que não passou na comparação (pode até usar o expect(..).to.be.deep.eq(..) para comparar o objeto inteiro)

Reply:
até pq nessa abordagem vc vai precisar fazer umas gambiarras para testar outros cenários:
para testar um usuário com senha errada iria precisar criar outro describe só pra isso ou fazer algum vodoo aqui nesse describe

Reply:
Entendi. No meu último estágio eu utilizei **RSpec** e **Jest** e a empresa seguia o padrão de ter apenas uma asserção por `it()`. Acabei seguindo esse costume (e realmente tive que fazer uma gambiarras em alguns casos).